Drafts download and deployment docs for ELSER

szabosteve opened the pull request:
## Overview

This PR adds docs on how to download and deploy ELSER.

### Preview

[ELSER](https://stack-docs_2371.docs-preview.app.elstc.co/guide/en/machine-learning/master/ml-nlp-elser.html)

szabosteve commented on docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc:
```diff
@@ -7,4 +7,116 @@
++
+--
+[role="screenshot"]
+image::images/ml-nlp-elser-deployment.png[alt="Deploying ELSER",align="center"]
```
```suggestion
image::images/ml-nlp-deploy-elser.png[alt="Deploying ELSER",align="center"]
```

droberts195 commented on docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc:
```diff
@@ -7,4 +7,116 @@
 
-[TBD]
\ No newline at end of file
+Elastic Learned Sparse Encoder - or ELSER - is a large language model (LLM) 
```
Please add the tech preview banner to this page.

qherreros reviewed (commented):
Very nice work! I made a small suggestion on ELSER.

droberts195 commented on docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc:
```diff
@@ -7,4 +7,108 @@
+== Download ELSER
+
+You can download ELSER either from the Trained Models page in {kib} or by using 
+the Dev Console.
```
Also from Enterprise Search - see https://github.com/elastic/kibana/pull/156080.

I am not sure how heavily we should push people towards the Enterprise Search app in the ML docs, but Enterprise Search is where we're planning to make the semantic search experience as friendly as possible, so we should certainly mention it here.

szabosteve replied:
I'm adding a section that covers this method, too.

davidkyle commented on docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc:
```diff
@@ -7,4 +7,108 @@
\ No newline at end of file
+experimental[]
+
+Elastic Learned Sparse Encoder - or ELSER - is a text expansion model that uses 
+sparse vector representation and trained by Elastic . As a retrieval model, 
```
```suggestion
Elastic Learned Sparse Encoder - or ELSER - is a representational model trained by Elastic that creates a
sparse vector representation  . As a retrieval model, 
```

Text expansion the _user friendly_ name we have chosen for the search query but it isn't a common term that people will recognise. I like @qherreros description of expanding text, maybe use the 'text expansion' after that

szabosteve commented:
@qherreros @droberts195 @davidkyle I updated the PR with your suggestions. In [8ac8269](https://github.com/elastic/stack-docs/pull/2371/commits/8ac82693ef28ea69b3e580b1e347ea7841794e01), I added a new section that covers the steps of deploying the model from the Enterprise Search app.

qherreros reviewed (approved):
LGTM !

droberts195 commented on docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc:
```diff
@@ -9,6 +9,125 @@
+[[download-elser]]
+== Download ELSER
+
+You can download ELSER either from the **{ml-app}** > **Trained Models** page or 
+by using the Dev Console.
```
```suggestion
You can download ELSER either from the **{ml-app}** > **Trained Models** page, 
by using the Dev Console, or using {ents}.
```

szabosteve replied:
@droberts195 If I understand [this PR](https://github.com/elastic/kibana/pull/156080) correctly, you need to have ELSER downloaded in order to deploy the model through the Enterprise Search application. This is the reason why I didn't mention Enterprise Search here, only in the section about deploying the model.

szabosteve replied:
Okay, I modified in [94d47dd](https://github.com/elastic/stack-docs/pull/2371/commits/94d47dda64547413fa17e280efaa30988499e8cc).

demjened replied:
> it's possible to both download and deploy ELSER just using the Enterprise Search app right?

Yes, correct.

davidkyle commented on docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc:
```diff
@@ -9,19 +9,137 @@
 experimental[]
 
+
+Elastic Learned Sparse Encoder - or ELSER - is a representational model trained 
```
```suggestion
Elastic Learned Sparse EncodeR - or ELSER - is a representational model trained 
```

For the R in ELSER. Is that correct @qherreros

qherreros replied:
Yes, true!

davidkyle reviewed (approved):
LGTM

droberts195 commented on docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc:
```diff
@@ -113,3 +231,11 @@ xpack.ml.model_repository: file://${path.home}/config/models/`
+
+[discrete]
+[[further-readings]]
+== Further readings
```
```suggestion
== Further reading
```

droberts195 reviewed (approved):
LGTM
